pslab/external/ccs811.py

```diff
@@ -0,0 +1,144 @@
+    _MEAS_MODE = 0x01
+    _ALG_RESULT_DATA = 0x02  # ALG_RESULT_DATA # R 8 bytes the next two bytes contain a ppb estimate of the total VOC level. Raw ADC data values for resistance and current source
+    _RAW_DATA = 0x03  # RAW_DATA # R 2 bytes used. Temperature and humidity data can be written to
+    _ENV_DATA = 0x05  # ENV_DATA # W 4 bytes enable compensation Thresholds for operation when interrupts are only
+    _THRESHOLDS = 0x10  # THRESHOLDS # W 4 bytes generated when eCO2 ppm crosses a threshold The encoded current baseline value can be read. A
+    # BASELINE # R/W 2 bytes previously saved encoded baseline can be written.
+    _BASELINE = 0x11
+    _HW_ID = 0x20  # HW_ID # R 1 byte Hardware ID. The value is 0x81
+    _HW = 0x21  # HW Version # R 1 byte Hardware Version. The value is 0x1X Firmware Boot Version. The first 2 bytes contain the
+    _FW_Boot_Version = 0x23  # FW_Boot_Version # R 2 bytes firmware version number for the boot code. Firmware Application Version. The first 2 bytes contain
+    # FW_App_Version # R 2 bytes the firmware version number for the application code
+    _FW_App_Version = 0x24
+    # Internal_State # R 1 byte Internal Status register Error ID. When the status register reports an error its
+    _Internal_State = 0xA0
+    # ERROR_ID # R 1 byte source is located in this register If the correct 4 bytes ( 0x11 0xE5 0x72 0x8A) are written
+    _ERROR_ID = 0xE0
+    # SW_RESET # W 4 bytes to this register in a single sequence the device will reset and return to BOOT mode.
+    _SW_RESET = 0xFF
+
+    # Figure 25: CCS811 Bootloader Register Map
+    # Address Register R/W Size Description
+    _STATUS = 0x00
+    _HW_ID = 0x20
+    _HW_Version = 0x21
+    _APP_ERASE = 0xF1
+    _APP_DATA = 0xF2
+    _APP_VERIFY = 0xF3
+    _APP_START = 0xF4
+    _SW_RESET = 0xFF
+
+    # Bit(s) Field
+    _REG_STATUS_BIT_FW_MODE = 7
+    _REG_STATUS_BIT_APP_ERASE = 6
+    _REG_STATUS_BIT_APP_VERIFY = 5
+    _REG_STATUS_BIT_APP_VALID = 4
+    _REG_STATUS_BIT_DATA_READY = 3
+    # 2:1 -
+    _REG_STATUS_BIT_ERROR = 0
+
+    def __init__(self, **args):
+        self._ADDRESS = args.get('address', self._ADDRESS)
+        super().__init__(self._ADDRESS)
+        self.fetchID()
+        self.softwareReset()
+
+    def softwareReset(self):
+        self.write([0x11, 0xe5, 0x72, 0x8a], self._SW_RESET)
+
+    def fetchID(self):
+        hardware_id = (self.read(1, self._HW_ID))[0]
+        print(f'hex(hardware_id) = {hex(hardware_id)}')
+        time.sleep(0.02)  # 20ms
+        hardware_version = (self.read(1, self._HW_Version))[0]
+        print(f'hex(hardware_version) = {hex(hardware_version)}')
+        time.sleep(0.02)  # 20ms
+        boot_version = (self.read(2, self._FW_Boot_Version))[0]
+        print(f'hex(boot_version) = {hex(boot_version)}')
+        time.sleep(0.02)  # 20ms
+        app_version = (self.read(2, self._FW_App_Version))[0]
+        print(f'hex(app_version) = {hex(app_version)}')
```

Comment:
These values should be returned, not printed.

Reply:
Fixed in 6b00a77